### DISCUSSION TITLE
Subscription type after the creation of the subscription using Pulsar Admin createSubscription method

### DISCUSSION THREAD

**Discussion by alexesom** [apache/pulsar#22339] [Q&A] [answered]
Hello!

I found that after creating a subscription using the [createSubscription method](https://github.com/apache/pulsar/blob/567174f43528c0f7ae917bfb5166213973c62c29/pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java#L1834) from PulsarAdmin, the type of the subscription is `null`. Is the type of the subscription automatically assigned after a consumer subscribes to a newly created subscription?

Thanks in advance! :slightly_smiling_face:

**Comment by visortelle** [comment, marked as answer]
@alexesom I think it's the correct answer to your question :)

https://stackoverflow.com/a/75485553/4182882

> The subscription type is determined when the first consumer connects. Pulsar will not persist the subscription type to the metadata store, so Pulsar doesn't have APIs to create a subscription with the subscription type (The created subscription is maintained in the metadata store).
And if the subsequent consumers are using different subscription types, they will fail when they subscribe to the topic. After all consumers disconnect, the subscription will change to "No subscription type" state, so the subscription gets a chance to switch to a different subscription type as before.
